### ISSUE TITLE
Editor crash when closing empty "modified" scene with keyboard shortcut

### ISSUE THREAD

**Issue by Cykyrios** [godotengine/godot#46959]
**Godot version:**
3.2.3 stable, also happens on custom build 3.2.4.rc.custom_build.b169a16cb

**OS/device including version:**
Linux Mint 20 Cinnamon

**Issue description:**
The editor crashes when you try to close an empty scene that is marked as modified, but this only happens if you press the Ctrl+Shift+W shortcut. This can typically happen after changing project settings.

**Steps to reproduce:**
Open any project, or create a new project.
Close any open scenes.
Go to project settings and change settings so that the scene tab is marked as modified (*).
Close scene with keyboard shortcut Ctrl+Shift+W.

**Minimal reproduction project:**
Not relevant.

Below is the crash backtrace (from my 3.2.4 custom build, as 3.2.3 did not output anything useful):
```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7fe9bfdd4210] (??:0)
[2] CowData<wchar_t>::_ref(CowData<wchar_t> const&) (godot/./core/cowdata.h:365)
[3] String::String(String const&) (godot/./core/ustring.h:357)
[4] Node::get_filename() const (godot/scene/main/node.cpp:1895)
[5] EditorNode::_menu_option_confirm(int, bool) (godot/editor/editor_node.cpp:2344)
[6] EditorNode::_menu_option(int) (godot/editor/editor_node.cpp:1061)
[7] MethodBind1<int>::call(Object*, Variant const**, int, Variant::CallError&) (godot/./core/method_bind.gen.inc:775 (discriminator 12))
[8] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (godot/core/object.cpp:919 (discriminator 1))
[9] Object::emit_signal(StringName const&, Variant const**, int) (godot/core/object.cpp:1246 (discriminator 1))
[10] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/core/object.cpp:1304)
[11] PopupMenu::activate_item(int) (godot/scene/gui/popup_menu.cpp:1198 (discriminator 4))
[12] PopupMenu::activate_item_by_event(Ref<InputEvent> const&, bool) (godot/scene/gui/popup_menu.cpp:1132)
[13] MenuButton::_unhandled_key_input(Ref<InputEvent>) (godot/scene/gui/menu_button.cpp:47 (discriminator 6))
[14] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (godot/./core/method_bind.gen.inc:775 (discriminator 12))
[15] Object::call_multilevel(StringName const&, Variant const**, int) (godot/core/object.cpp:761 (discriminator 1))
[16] SceneTree::_call_input_pause(StringName const&, StringName const&, Ref<InputEvent> const&) (godot/scene/main/scene_tree.cpp:952)
[17] Viewport::unhandled_input(Ref<InputEvent> const&) (godot/scene/main/viewport.cpp:2935 (discriminator 2))
[18] Viewport::_vp_unhandled_input(Ref<InputEvent> const&) (godot/scene/main/viewport.cpp:1505)
[19] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (godot/./core/method_bind.gen.inc:775 (discriminator 12))
[20] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (godot/core/object.cpp:919 (discriminator 1))
[21] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/core/object.cpp:846)
[22] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/scene/main/scene_tree.cpp:276)
[23] SceneTree::input_event(Ref<InputEvent> const&) (godot/scene/main/scene_tree.cpp:449 (discriminator 6))
[24] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (godot/main/input_default.cpp:429)
[25] InputDefault::parse_input_event(Ref<InputEvent> const&) (godot/main/input_default.cpp:273)
[26] InputDefault::flush_accumulated_events() (godot/main/input_default.cpp:692)
[27] OS_X11::process_xevents() (godot/platform/x11/os_x11.cpp:2337)
[28] OS_X11::run() (godot/platform/x11/os_x11.cpp:3639)
[29] godot/bin/godot.x11.tools.64(main+0x125) [0x16f9a7b] (godot/platform/x11/godot_x11.cpp:57)
[30] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fe9bfdb50b3] (??:0)
[31] godot/bin/godot.x11.tools.64(_start+0x2e) [0x16f989e] (??:?)
-- END OF BACKTRACE --
```

**Comment by hilfazer** [comment]
Same on Windows 7. 3.2.3 stable and 3.2.4 RC4.

**Comment by jmb462** [comment]
Also occurs in 4.0 master branch.

Fix proposal : #46970
